### PR TITLE
yarn-lock-converter: init at 0.0.2

### PR DESCRIPTION
**Pull request by gador**
###### Description of changes

https://www.npmjs.com/package/@vht/yarn-lock-converter?activeTab=readme 

Allows to convert v2 and v3 yarn.lock files to convert to v1 lockfiles. 

EDIT:

~~Yes, I've seen https://github.com/NixOS/nixpkgs/issues/229475. Unfortunately upstream only provides a `yarn.lock` file and no `package-lock.json` file.~~
(see below)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Please see #229475 how to package new cli tools.

**Comment by gador** [comment]
> Please see #229475 how to package new cli tools.

I've seen this PR. I even mentioned it in the description. Please see my comment there.

File: pkgs/tools/package-management/yarn-lock-converter/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+let
+  source = lib.importJSON ./source.json;
+in
+buildNpmPackage rec {
+  pname = "yarn-lock-converter";
+  inherit (source) version;
+
+  src = fetchurl {
+    url = "https://registry.npmjs.org/@vht/yarn-lock-converter/-/yarn-lock-converter-${version}.tgz";
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/yarn-lock-converter/default.nix:17]: We should add a fetcher for this.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 231687` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>yarn-lock-converter</li>
</ul>
</details>

**Pull request merged**